### DISCUSSION TITLE
Spoon-feeding instructions to install Copilot in Neovim with LazyVim?

### DISCUSSION THREAD

**Discussion by b9Joker108** [community/community#84195] [Copilot Conversations]
I am coding in Termux on Android. Is there an easy way to plugin Copilot into Neovim with LazyVim? I am trying to setup an IDE in Termux. I am having some troubles working through some Lua syntax in my init.lua file, so I cannot currently edit with Neovim. But hopefully, I will soon work through these issues. When I do, I would like to plugin Copilot into the mix if that is possible. I could find no express instruction on how to realise this through the documentation, and request assistance. Thanks for your time!

**Comment by trustieee** [comment]
Copilot is built in as an extra plugin to lazyvim. You can enable it by opening up `:LazyExtras` and navigating to `coding.copilot`, and enabling it.

**Reply by b9Joker108** [reply]
Thanks for your response. I finally got my init.lua file Lua syntax correct and Lazy is loading my plugins. I tried running: ’:LazyExtras’ in Neovim in Termux, and I get the error output: E492: Not an editor command: LazyExtras. I tried searching GitHub for "LazyExtras" and it returned nothing. I also tried various permutations of Googling it, but no cigar. I was wondering if you could please be more specific.

**Reply by trustieee** [reply]
https://www.lazyvim.org/extras/coding/copilot

**Reply by laurensdc** [reply]
> I tried running: ’:LazyExtras’ in Neovim in Termux, and I get the error output: E492: Not an editor command: LazyExtras. I tried searching GitHub for "LazyExtras" and it returned nothing. I also tried various permutations of Googling it, but no cigar. I was wondering if you could please be more specific.

Hey @b9Joker108 , I'm having the exact same thing you described and can't find any information to help me out. Did you resolve this?

Because when I run `:LazyExtras` I also get the `E492: Not an editor command: LazyExtras`

And when importing an extra plugin like CoPilot, it fails.
<img width="1030" alt="image" src="https://github.com/community/community/assets/17620692/1a2544a9-cb1c-4e78-8942-832bd266a26d">

I see that @b9Joker108 also used Kickstart.nvim like me, maybe the LazyVim setup isn't correct?

**Reply by laurensdc** [reply]
I got it to work when I found [this closed issue](https://github.com/nvim-lua/kickstart.nvim/issues/184) on kickstart.nvim.

In `init.lua`, I could add Copilot like this:

```lua
require('lazy').setup({
  -- NOTE: Plugins can be added with a link (or for a github repo: 'owner/repo' link).
  'tpope/vim-sleuth', -- Detect tabstop and shiftwidth automatically
  'github/copilot.vim',
```

**Reply by b9Joker108** [reply]
@Laurensdc  Two different versions of kickstart.nvim were my first introduction to Lua and configuring Neovim and installing plugins. It was a learning curve and sometimes frustrating. I found Lua arcane and it is supposed to be simple, lightweight and efficient. From kickstarter, on whim, I then graduated to a distribution coded by Josean on GitHub, from a YouTube video he demonstrated coding it. Over about six (6) months, I tweaked and further customised this distribution, and used it for some Python scripting, but didn't use all the bells-and-whistles, the different plugins. But, I learnt much about Lua and Neovim and setting up and editing the Neovim Lua configuration files. But, for many reasons, I just found I still preferenced vanilla `vim` as my favoured editor, even though I had a taste of the potential editor 'god-mode' of a bespoke Neovim distribution. On discovering the brilliant Dotfyle website: https://dotfyle.com/ , I haven't looked back. I ended up nuking the Josean distribution and config, as its file structure was unnecessarily labyrinthine, idiosyncratic and somewhat non-standard. I had heard of Cursor, the new AI editor that is making waves and endeavoured installing an unofficial implementation/patch of it for aarch64. This was a disaster and I dispensed with it. I then searched for "Cursor-like Neovim plugins" and I came across the project: https://github.com/yetone/avante.nvim I then looked up this plugin in the Dotfyle website and there were a number of bespoke distributions that used it. I went with the distribution that had the most plugins (259) that included avante.nvim and used the Lazy package manager which I favour: https://dotfyle.com/b9Joker108 I didn't fork it. I just git cloned all of the person's dotfiles locally and I am going through their logic and wisdom and incorpating it into my dotfiles as appropriate and culling the files I don't need. Just the Neovim config, I extracted, and overwrote the Josean distribution spec in my GitHub repository, kept the kickstarter out of nostalgia and git pushed the new config into my repository. I am now customising and further developing this locally and have not yet updated my repository README.md to reflect these changes. I only installed this new Neovim distro two (2) days ago. I am far from knowing how to use it. But this time, I am going to make a point of slowly familiarising myself with each plugin and its purpose and functionality. Dotfyle generates an excellent Markdown report of all the plugins in your configuration/distribution and this is an invaluable resource to help support familiarisation with each plugin, its functionality and usage. All of this excessive verbiage, is really just to say: use Dotfyle!
